lib/runvxenlight/xenlight-runv.go

```diff
@@ -124,14 +124,15 @@ void runvxl_childproc_setmode(libxl_ctx *ctx) {
     libxl_childproc_setmode(ctx, &childproc_hooks, 0);
 }
 
-int runvxl_add_nic(libxl_ctx *ctx, int32_t domid, int id, char *bridge, char *mac) {
+int runvxl_add_nic(libxl_ctx *ctx, int32_t domid, int id, char *bridge, char *device, char *mac) {
     libxl_device_nic nic;
     int ret;
 
     libxl_device_nic_init(&nic);
     nic.bridge = strdup(bridge);
     nic.nictype = LIBXL_NIC_TYPE_VIF;
     nic.devid = id;
+    nic.ifname = strdup(device);
```

Comment:
where is the code that nic.ifname and nic.bridge are to be freed?
Is it libxl_device_nic_dispose(&nic); ?

Reply:
yes